### PR TITLE
enhance EventStream and Kubevirt events page

### PR DESCRIPTION
**Pull request by suomiy**
This adds a possibility to filter and see events of multiple resources in VM events page.

example: 
evento

### CODE REVIEW

**Comment by spadgett** [comment]
I'm not necessarily against this change, but it's inconsistent with other events tabs. For instance, we don't show pod events when looking at a replica set. I think we should consider this change across all console pages instead of making it in one place.

**Comment by suomiy** [comment]
It is quite useful to have all relevant events in one place. Especially for the `VirtualMachineInstance` object, because we are not planning to display it as a standalone entity.

Could we start with VM list first and then decide which other events to enhance in the followups? + this PR should not introduce any regressions

**Comment by jelkosz** [comment]
@spadgett the thing with VMs is that there are many supporting pods around a VM which start and than die when they finish performing something. From user's PoV those might be just actions (e.g. migrate a VM - this creates a migration pod. Start a VM - this creates a VMI instance which in turn creates a pod in which the qemu process lives, once there will be snapshots, the snapshotting a VM will create a snapshot pod, we have some CDI pods (cloning / importing disks from URL/container/whatever) etc..). Having one place where you have a holistic view showing what is going on with the VM is extremely useful.
I see value of this also for other entities, but my (probably biased :) ) take on it is that it is crucial for VMs and nice to have for other places. So I would really like to start here.

**Comment by cloudbehl** [comment]
The event filtering is something that is also required by the storage dashboards. As we want to show all the resource in the events tab that is related to the ceph/nooba storage.

**Comment by vojtechszocs** [comment]
> @spadgett the thing with VMs is that there are many supporting pods around a VM which start and than die when they finish performing something. From user's PoV those might be just actions (e.g. migrate a VM - this creates a migration pod. Start a VM - this creates a VMI instance which in turn creates a pod in which the qemu process lives, once there will be snapshots, the snapshotting a VM will create a snapshot pod, we have some CDI pods (cloning / importing disks from URL/container/whatever) etc..). Having one place where you have a holistic view showing what is going on with the VM is extremely useful.
> I see value of this also for other entities, but my (probably biased :) ) take on it is that it is crucial for VMs and nice to have for other places. So I would really like to start here.

Thanks for this clarification - the essence is that VMs in k8s aren't represented by a single CRD but rather by multiple CRDs - VM (metadata), VM-Instance, VM-Instance-ReplicaSet, VM-Instance-Migration, etc.

This is in contrast with the traditional approach of listing / detailing / eventing on a single CRD kind (like Pod model) which can be mapped to a single logical object (like the k8s concept of Pod).

**Comment by matthewcarleton** [comment]
+1 I think this makes a lot of sense and would be beneficial to see across the entire console. Starting with VMs seems reasonable to me.

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    name: vmName,
+    namespace: vmNamespace,
+  };
+
+  const vmFilter = (obj) => _.isMatch(obj, { ...vmCompareObject, kind: VirtualMachineModel.kind });
+  const vmiFilter = (obj) =>
+    _.isMatch(obj, { ...vmCompareObject, kind: VirtualMachineInstanceModel.kind });
+
+  const launcherPodFilter = ({ kind, namespace, name }) =>
+    kind === PodModel.kind &&
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx]: I'd extract these functions as utilities, like `isVM`, `isVMI`, `isLauncherPod`, `isImporterPod` and `isMigration`.

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx]: extract into its own filter functions

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { VMTabProps } from './types';
+import {
+  importerPodEventFilter,
+  launcherPodEventFilter,
+  vmEventFilter,
+  vmiEventFilter,
+  vmiMigrationEventFilter,
+} from '../../selectors/event';
+
+export const VmEvents: React.FC<VMTabProps> = ({ obj: vm }) => (
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-events.tsx]: Nit: `VMEvents`

File: frontend/packages/kubevirt-plugin/src/selectors/event/filters.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { getName, getNamespace } from '@console/shared/src';
+import * as _ from 'lodash';
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/src/selectors/event/filters.ts]: Nit: vendor imports should come first.

File: frontend/packages/kubevirt-plugin/src/selectors/event/filters.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import { PodModel } from '@console/internal/models';
+import { VMIKind, VMKind } from '../../types/vm';
+import {
+  VirtualMachineInstanceMigrationModel,
+  VirtualMachineInstanceModel,
+  VirtualMachineModel,
+} from '../../models';
+import { VIRT_LAUNCHER_POD_PREFIX } from '../../constants/vm';
+
+export const vmEventFilter = (vm: VMKind) => {
```
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/src/selectors/event/filters.ts]: Great to see those filters extracted into a separate module.

**Comment by suomiy** [comment]
fixed nitpicks

**Pull request merged**